### ISSUE TITLE
RCS1244 false-positive in statement lambdas

### ISSUE THREAD

**Issue by jhinder** [JosefPihrt/Roslynator#957]
**Product and Version Used**: Roslynator 4.1.1 in VS 17.3.5

**Steps to Reproduce**:
Apply the codefix for RCS1244 on the `default()` expression.

```csharp
var x = () =>
{
    return default(object);
               // ~~~~~~~~ RCS1244: Simplify 'default' expression
};
```

This affects statement lambdas, but not expression lambdas.
```csharp
var x = () => default(object); // No diagnostics
```

**Actual Behavior**:
The argument is removed, leaving a typeless default literal.
```csharp
var x = () =>
{
    return default;
        // ~~~~~~~ CS8917: There is no target type for the default literal
};
```

**Expected Behavior**:
The codefix is not offered.
